### DISCUSSION TITLE
Can Api definitions, version definition, enable_unprivileged_ports and enable_unprivileged_icmp definitions explain why rke2-canal pods keep restarting after a node reboot ?

### DISCUSSION THREAD

**Discussion by marcredhat** [rancher/rke2#5079] [Q&A]
The last part of 
https://medium.com/@ozkanpoyrazoglu/troubleshooting-kube-proxy-and-rke2-canal-pods-restart-loop-lessons-from-upgrading-ubuntu-20-04-7ac6107c6475 
seems to suggest that 
not setting
`Api definitions, version definition, enable_unprivileged_ports and enable_unprivileged_icmp definitions` 
can explain why rke2-canal pods keep restarting after a node reboot.
 
Configs shown at https://github.com/rancher/rke2/issues/3385 also set 
`enable_unprivileged_ports` and `enable_unprivileged_icmp`.

Could you please provide the details on the role of the above flags? 

@brandond

**Comment by brandond** [comment]
I can't provide any insight into that user's motivations in customizing their containerd config template to modify those values. That shouldn't be necessary, as they are enabled by default if the kernel supports it. I suspect their problems may have been self-infliced, as their template does not appear to actually contain any templating values, but is instead just a very basic flat config file.

* https://github.com/k3s-io/k3s/blob/3f237230350b5170eef4e54c7826d88433182efc/pkg/agent/templates/templates_linux.go#L20-L21
* https://github.com/k3s-io/k3s/blob/3f237230350b5170eef4e54c7826d88433182efc/pkg/agent/containerd/config_linux.go#L73

If you're having a specific problem with your nodes, please create a new issue and fill out the issue template. If you want to discuss blog posts, that might be better handled in a discussion, instead of an issue.
